crowbar_framework/lib/crowbar/backup/restore.rb

```diff
@@ -42,7 +43,15 @@ def restore
           if any_errors?
             cleanup
             @backup.errors.add(:restore, error_messages.join(" - "))
-            Thread.exit if @thread
+            Rails.logger.error("Restore failed: #{@backup.errors.full_messages.first}")
+            if @thread
+              Rails.logger.debug(
+                "Exiting thread: #{@thread.inspect} \
+                Backtrace: #{@thread.backtrace} \
+                Alive: #{@thread.alive?}"
```

Comment:
I'm not sure the backtrace and alive bits would bring us something we don't already know (or am I missing something)? I would keep the "exiting thread" bit -- but maybe just have it be "Exiting restore thread due to failure".